CNTN6.py

```diff
@@ -0,0 +1,88 @@
+class BiologicalSequence:
+    def __init__(self, sequence):
+        self.sequence = sequence
+        self.ALPHABET = {'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y'}
+
+    def __len__(self):
+        return len(self.sequence)
+
+    def __getitem__(self, idx):
+        return self.sequence[idx]
+
+    def __get_slice__(self, start, end):
+        if start < 0 or start > len(self.sequence):
+            raise IndexError("start index out of range")
+
+        if end < 0 or end > len(self.sequence):
+            raise IndexError("end index out of range")
+
+        if start > end:
+            raise ValueError("start index must be less than or equal to end index")
+
+        return self.sequence[start:end]
+
+    def __str__(self):
+        return self.sequence
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}({self.sequence!r})'
+
+    def is_valid_bioseq(self):
+        for char in self.sequence:
+            if char not in self.ALPHABET:
+                return False
+        return True
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    def complement(self, COMPLEMENT_NUCLEOTIDES=None):
+        if COMPLEMENT_NUCLEOTIDES is None:
+            COMPLEMENT_NUCLEOTIDES = {'A': 'T', 'T': 'A', 'C': 'G', 'G': 'C'}
+
+        complement_seq = ''
+        for nucleotide in self.sequence:
+            complement_seq += COMPLEMENT_NUCLEOTIDES.get(nucleotide)
+        return complement_seq
+
+    def gc_content(self):
+        return (self.sequence.count('G') + self.sequence.count('C'))/len(self.sequence)
+
+
+class RNASequence(NucleicAcidSequence):
+    REV_TRANSCRTPTION_COMPLEMENT = {'A': 'T', 'U': 'A', 'C': 'G', 'G': 'C'}

```

Comment:
о! Класс!

Reply:
алфавит тоже сюда можн перенести)
```suggestion
    REV_TRANSCRTPTION_COMPLEMENT = {'A': 'T', 'U': 'A', 'C': 'G', 'G': 'C'}
    ALPHABET = {'A','U','G','C'}
```